test/lld/gdollar_mainmodule.wat.out

```diff
@@ -137,27 +79,22 @@
    )
   )
   (global.set $gimport$8
-   (call $g$_ZN20__em_asm_sig_builderI19__em_asm_type_tupleIJiEEE6bufferE)
+   (call $g$_ZN20__tester_sig_builderI19__tester_type_tupleIJiEEE6bufferE)
   )
  )
 )
 (;
 --BEGIN METADATA --
 {
-  "asmConsts": {
-    "0": ["{ Module.print(\"Hello world\"); }", ["iii"], [""]],
-    "34": ["{ return $0 + $1; }", ["iii"], [""]],
-    "57": ["{ Module.print(\"Got \" + $0); }", ["iii"], [""]]
-  },
   "staticBump": 4294966728,
   "tableSize": 0,
   "initializers": [
     "__assign_got_enties",
     "__wasm_call_ctors"
   ],
   "declares": [
-    "g$_ZN20__em_asm_sig_builderI19__em_asm_type_tupleIJEEE6bufferE",
-    "g$_ZN20__em_asm_sig_builderI19__em_asm_type_tupleIJiEEE6bufferE"
+    "g$_ZN20__tester_sig_builderI19__tester_type_tupleIJEEE6bufferE",
+    "g$_ZN20__tester_sig_builderI19__tester_type_tupleIJiEEE6bufferE"
```

Comment:
Is there a reason you don't want to include the C source code? That way we can use `scripts/test/generate_lld_tests.py` to regenerate, and we can see the type of input that triggers this.

I feel like it would be a win win for readability and understandability of what we are testing here.

Also, if you write the code in C rather than C++ it would make these symbols easier to read.

Reply:
(All the other tests in the this directory have associated C/C++ source code).

Reply:
I seems more readable to me to read the wat directly, I guess you disagree? With the wat I can see exactly that I have what the pass works on, directly...

I renamed the symbols now, good idea. Also I realized I can remove the extra main() and other stuff. Now it's less than 20 lines - I think much more readable than the other testcases. What do you think?

Reply:
I like to see that wat too and have the checked in, but I think its important for wasm-emscripten-finalize to see when/how a given wat file is produced by the compiler.  The input of wasm-emscripten-finalize is always the output of llvm so we really only case about things the llvm can produce.

I do agree this test case is now very simple and readable to I won't block it.

Reply:
I think the problem is that I have no way to be sure that the wat you are now checking in is similar to what llvm produces.. I can tell that you are testing something.. but your edits mean that I'm not sure if its the exact right thing.   So I'm a little less confident as a reviewer of this change that its actually doing the right thing.  

I don't know how complex a minimal C example would be though.. maybe it would be much harder to read the wat file produced by lld?

Reply:
That's a good point, yeah, we are at risk of the testcase eventually becoming different from what lld emits. We do have end-to-end testing in the emscripten repo (we'd see `g$` changes in the metadce tests) so I'm not worried. But maybe that's an argument for having both types of tests here.